### DISCUSSION TITLE
Full width message bubbles

### DISCUSSION THREAD

**Discussion by 0xCA** [Taewan-P/gpt_mobile#113] [Ideas]
Please add an option to enable full with messages, there is a significant amount of space that is unused. Especially, with nested lists in markdown responses. This can either widen message bubbles to fit, or remove bubbles completely, separating messages by a line separator or color.

**Comment by Taewan-P** [comment]
While it may seem that it is using a lot of space, the bubble width is calculated by the screen width subtracted by some margins... If you think the bubble width is small, this means that your phone width is small. I will see what I can do to improve this UX though.

**Reply by ProjectMoon** [reply]
With a Pixel 7 Pro, I have a pretty big chunk of space left off to the side. What device are you using as a reference for testing the width?

**Reply by Taewan-P** [reply]
I use Pixel 8 Pro & Pixel 9 Pro XL to test the app.
Can you show me a screenshot? What kind of big space is at the left side?

**Reply by ProjectMoon** [reply]
I'd say that the bubble takes up about 75% to 80% screen width. It's not a problem for regular text, but nested lists etc could make use of extra space.

**Reply by Taewan-P** [reply]
How is your Display and Text settings? 

![Screenshot_20241118-211442](https://github.com/user-attachments/assets/e8668c22-3015-4044-b7f0-e7cba13df544)

Display size can effect the bubble size I guess...

**Reply by ProjectMoon** [reply]
Here's an example of what it looks like on my screen. I feel like the right side could expand further.
![Screenshot_20241118-145525](https://github.com/user-attachments/assets/a41f001e-405c-4277-b825-10dfcbb3855a)

**Comment by 0xCA** [comment]
<details>
<summary>To illustrate what I mean, here are two screenshots: from the app and from the telegram</summary>

![Screenshot_1](https://github.com/user-attachments/assets/d17165de-d950-4977-b244-4efac31e9722)
![Screenshot_2](https://github.com/user-attachments/assets/5d23cbd3-95cd-426f-9c7c-ad63210f5efd)

</details>

Telegram is slightly bad at markdown, but you can see the idea. With more or less identical font, telegram offers significantly more density.
When the response includes code blocks, they are _very_ hard to read in such a vertical form. They need full width the most.

<details>
<summary>Also, this spacing should be moved to better group info blocks</summary>

![Screenshot](https://github.com/user-attachments/assets/66bad5e6-aecc-4d33-b91c-72efbb503793)

</details>

**Reply by Taewan-P** [reply]
For markdown spacing, please leave the issue [here](https://github.com/jeziellago/compose-markdown). I cannot modify that part in my app.

**Comment by Taewan-P** [comment]
As requested, I am working on implementing full width chat bubbles.. Please wait for upcoming updates, thanks.

**Comment by Taewan-P** [comment]
Hello, this feature is now implemented in [the latest version](https://github.com/Taewan-P/gpt_mobile/releases). Thanks!
cc @0xCA @ProjectMoon
